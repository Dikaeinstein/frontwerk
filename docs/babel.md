# frontwerk 🛠️ builds with BabelJS

_Details coming soon_

[Read the source](https://github.com/tricinel/frontwerk/blob/master/src/scripts/babel.js)
