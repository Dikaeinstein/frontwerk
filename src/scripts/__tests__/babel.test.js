/* eslint-disable global-require */
jest.mock('cross-spawn');
jest.mock('jest');

const cases = require('jest-in-case');

const testCases = [
  {
    name: 'calls babel CLI with default args'
  },
  {
    name: '--no-copy-files will not copy the files',
    args: ['--no-copy-files']
  },
  {
    name: '--out-dir will use the specified out dir',
    args: ['--out-dir', './.tmp/build']
  },
  {
    name: '--no-clean will not clean the outDir',
    args: ['--no-clean']
  },
  {
    name: '--no-color will disable colored output',
    args: ['--no-color']
  },
  {
    name: 'does not use built-in config with .babelrc file',
    fileExists: filename => filename === '.babelrc'
  },
  {
    name: 'does not use built-in config with babel pkg prop',
    hasPkgProp: prop => prop === 'babel'
  },
  {
    name: 'does not use built-in ignore with --ignore',
    args: ['--ignore', '__custom__']
  },
  {
    name: 'does not use built-in config with --presets',
    args: ['--presets', './custom-presets.js']
  }
];

const testFn = ({
  fileExists = () => false,
  hasPkgProp = () => false,
  args = []
}) => {
  const { sync: crossSpawnSyncMock } = require('cross-spawn');
  const originalExit = process.exit;
  const originalArgv = process.argv;
  process.exit = jest.fn();

  Object.assign(require('../../utils/fileExists'), { fileExists });
  Object.assign(require('../../utils/pkg'), { hasPkgProp });
  Object.assign(require('../../utils/resolveBin'), {
    resolveBin: (modName, { executable = modName } = {}) => executable
  });

  try {
    // tests
    process.argv = ['node', '../babel', ...args];
    crossSpawnSyncMock.mockClear();

    require('../babel');

    expect(crossSpawnSyncMock).toHaveBeenCalledTimes(1);
    const [firstCall] = crossSpawnSyncMock.mock.calls;
    const [script, calledArgs] = firstCall;
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot();
  } catch (error) {
    throw error;
  } finally {
    // afterEach
    process.exit = originalExit;
    process.argv = originalArgv;
    jest.resetModules();
  }
};

cases('babel', testFn, testCases);
