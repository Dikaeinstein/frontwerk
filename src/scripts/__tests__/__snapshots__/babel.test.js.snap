// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format --no-clean will not clean the outDir 1`] = `"babel --out-dir dist --copy-files --ignore __tests__,__mocks__ --presets /Users/tricinel/Work/Personal/repos/ff-scripts/packages/ff-scripts/src/config/babelrc.js src --no-clean"`;

exports[`format --no-color will disable colored output 1`] = `"babel --out-dir dist --copy-files --ignore __tests__,__mocks__ --presets /Users/tricinel/Work/Personal/repos/ff-scripts/packages/ff-scripts/src/config/babelrc.js src --no-color"`;

exports[`format --no-copy-files will not copy the files 1`] = `"babel --out-dir dist --ignore __tests__,__mocks__ --presets /Users/tricinel/Work/Personal/repos/ff-scripts/packages/ff-scripts/src/config/babelrc.js src --no-copy-files"`;

exports[`format --out-dir will use the specified out dir 1`] = `"babel --copy-files --ignore __tests__,__mocks__ --presets /Users/tricinel/Work/Personal/repos/ff-scripts/packages/ff-scripts/src/config/babelrc.js src --out-dir ./.tmp/build"`;

exports[`format calls babel CLI with default args 1`] = `"babel --out-dir dist --copy-files --ignore __tests__,__mocks__ --presets /Users/tricinel/Work/Personal/repos/ff-scripts/packages/ff-scripts/src/config/babelrc.js src"`;

exports[`format does not use built-in config with .babelrc file 1`] = `"babel --out-dir dist --copy-files --ignore __tests__,__mocks__ src"`;

exports[`format does not use built-in config with --presets 1`] = `"babel --out-dir dist --copy-files --ignore __tests__,__mocks__ src --presets ./custom-presets.js"`;

exports[`format does not use built-in config with babel pkg prop 1`] = `"babel --out-dir dist --copy-files --ignore __tests__,__mocks__ src"`;

exports[`format does not use built-in ignore with --ignore 1`] = `"babel --out-dir dist --copy-files --presets /Users/tricinel/Work/Personal/repos/ff-scripts/packages/ff-scripts/src/config/babelrc.js src --ignore __custom__"`;
